packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -80,12 +80,59 @@ export 'package:flutter/services.dart' show
 
 // BIDIRECTIONAL TEXT SUPPORT
 
+/// An [InheritedElement] that has hundreds of dependencies but will
+/// infrequently change.  This provides a performance tradeoff where building
+/// the [Widget]s is faster but performing updates is slower.
+class _UbiquitousInheritedElement extends InheritedElement {
+  /// Creates an element that uses the given widget as its configuration.
+  _UbiquitousInheritedElement(InheritedWidget widget) : super(widget);
+
+  @override
+  void setDependencies(Element dependent, Object? value) {
+    if (value != null) {
```

Comment:
For my own understanding: Is the optimization here that we omit the bookkeeping that originally happened in this method? (Adding the dependent to a map?)

Reply:
Yep, that's correct.  You can see that in the profile I linked in the internal bug.

Reply:
Maybe leave that as a comment here for future readers, who want to understand why we did this.

Reply:
done